src/textbooks/Textbooks.jsx

```diff
@@ -36,54 +58,74 @@ const Textbooks = ({ courseId }) => {
-      <section className="mb-4 mt-5">
-        <SubHeader
-          title={intl.formatMessage(messages.headingTitle)}
-          breadcrumbs={<Breadcrumb ariaLabel={intl.formatMessage(messages.breadcrumbAriaLabel)} links={breadcrumbs} />}
-          headerActions={(
-            <Button
-              iconBefore={AddIcon}
-              // TODO: add handler for opening textbooks form
-              onClick={() => null}
-            >
-              {intl.formatMessage(messages.newTextbookButton)}
-            </Button>
-          )}
+    <>
+      <Container size="xl" className="px-4">
+        <section className="mb-4 mt-5">
+          <SubHeader
+            title={intl.formatMessage(messages.headingTitle)}
+            breadcrumbs={(
+              <Breadcrumb ariaLabel={intl.formatMessage(messages.breadcrumbAriaLabel)} links={breadcrumbs} />
+            )}
+            headerActions={(
+              <Button
+                iconBefore={AddIcon}
+                onClick={openTextbookForm}
+                disabled={isTextbookFormOpen}
+              >
+                {intl.formatMessage(messages.newTextbookButton)}
+              </Button>
+            )}
+          />
+          <Layout
+            lg={[{ span: 9 }, { span: 3 }]}
+            md={[{ span: 9 }, { span: 3 }]}
+            sm={[{ span: 12 }, { span: 12 }]}
+            xs={[{ span: 12 }, { span: 12 }]}
+            xl={[{ span: 9 }, { span: 3 }]}
+          >
+            <Layout.Element>
+              <article>
+                <section className="textbook-section">
+                  <div className="pt-4">
+                    {textbooks.length ? textbooks.map(({ tabTitle, chapters, id }) => (
+                      <TextbookCard
+                        key={id}
+                        chapters={chapters}
+                        title={tabTitle}
+                      />
+                    )) : (
+                      !isTextbookFormOpen && <EmptyPlaceholder onCreateNewTextbook={openTextbookForm} />
+                    )}
+                    {isTextbookFormOpen && (
+                      <TextbookForm
+                        closeTextbookForm={closeTextbookForm}
+                        initialFormValues={getTextbookFormInitialValues()}
+                        onSubmit={handleTextbookFormSubmit}
+                        onSavingStatus={handleSavingStatusDispatch}
+                        courseId={courseId}
+                      />
+                    )}
+                  </div>
+                </section>
+              </article>
+            </Layout.Element>
+            <Layout.Element>
+              <TextbookSidebar courseId={courseId} />
+            </Layout.Element>
+          </Layout>
+        </section>
+      </Container>
+      <ProcessingNotification
+        isShow={isShowProcessingNotification}
+        title={processingNotificationTitle}
+      />
+      <div className="alert-toast">
+        <InternetConnectionAlert
```

Comment:
[question] It looks like we don't handle 4xx and 5xx errors.
Can you mock response with an error from the `edx-platform` and check if the error actually appears.

Go to `cms/djangoapps/contentstore/rest_api/v1/views/textbooks.py` and change response to return Response(status=404) or another status error

Reply:
@monteri Error alert 4xx and 5xx should be discussed, because they do not appear.

Reply:
This should be done similar to https://github.com/raccoongang/frontend-app-course-authoring/blob/a1189d9a9103f2dc4a9d03bb603bd10f1526d712/src/course-unit/CourseUnit.jsx#L171

Reply:
Make sure this logic is the same as there. There it works already

Reply:
not working